### DISCUSSION TITLE
v0.12.15 - The smol anni update

### DISCUSSION THREAD

**Discussion by ThePotatoGuy** [Monika-After-Story/MonikaModDev#10084] [Announcements & Releases]
_goin on 6 years now..._

# Topic Changes

### Additions
* 6-year and 7-year anniversary dialogue
* new player bday poem

# Fixes

* typofixes

* fixed 1st year anniversary not stting love flag

* fixed introduction using narrator instead of Monika for some dialogue

* fixed window bugs in greetings back from <x>

# API Changes

* greeting/farewell probability rework - see #9725

* deprecations in function plugin stuff - see #9736

## What's Changed
* Improve greeting/fares probabilities code by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9725
* Deprecate bad function plugin arg by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9736
* Fix a missing `m` in introduction by @Wingdinggaster656 in https://github.com/Monika-After-Story/MonikaModDev/pull/9957
* [Drafts] - 6+7 Anniversaries by @lunulae in https://github.com/Monika-After-Story/MonikaModDev/pull/7324
* Add a missing `return "love"` by @Wingdinggaster656 in https://github.com/Monika-After-Story/MonikaModDev/pull/9944
* Fix greetings menu dlg by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9968
* Fix new annis by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/10081
* add new pbday poem from lewn by @multimokia in https://github.com/Monika-After-Story/MonikaModDev/pull/10083

**Full Changelog**: https://github.com/Monika-After-Story/MonikaModDev/compare/v0.12.14...v0.12.15

<hr /><em>This discussion was created from the release <a href='https://github.com/Monika-After-Story/MonikaModDev/releases/tag/v0.12.15'>v0.12.15 - The smol anni update</a>.</em>

**Comment by YamaruTF** [comment]
And yea.....Read buffer error....whenever i launch game it saying this:
[traceback.txt](https://github.com/Monika-After-Story/MonikaModDev/files/13265206/traceback.txt)
So....any help?

**Reply by JcLeChatones** [reply]
ik its been a year since this has been answered but also for me is also trace back errors when i try idk am i doing something wrong??

**Comment by William-Delancey** [comment]
how can i unlock the piano?
